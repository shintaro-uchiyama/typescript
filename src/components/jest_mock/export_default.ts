const returnString = "string from default";
export default returnString;
