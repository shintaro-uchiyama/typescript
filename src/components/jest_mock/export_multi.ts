export const returnString = "string from multi";

export const returnFunction = () => "function from multi";
