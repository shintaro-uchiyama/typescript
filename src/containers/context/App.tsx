import React from "react";
import { Content } from "../../components/context/Context";
import ThemeContext from "./theme-context";
import { themes } from "./themes";

interface State {
  theme: {
    background: string;
    foreground: string;
  };
  toggleTheme: () => void;
}

class App extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props);
    this.toggleThem = this.toggleThem.bind(this);
    // State also contains the updater function so it will
    // be passed down into the context provider
    this.state = {
      theme: themes.light,
      toggleTheme: this.toggleThem,
    };
  }

  public render() {
    // The entire state is passed to the provider
    return (
      <ThemeContext.Provider value={this.state}>
        <Content />
      </ThemeContext.Provider>
    );
  }

  private toggleThem() {
    this.setState({
      theme:
        this.state.theme === themes.dark
          ? themes.light
          : themes.dark,
    });
  }
}

export default App;
