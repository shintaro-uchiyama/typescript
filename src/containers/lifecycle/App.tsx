import React from "react";
import Lifecycle from "./Lifecycle";

interface State {
  id: string;
}
class App extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props);
    this.state = {
      id: "",
    };
    this.handleInput = this.handleInput.bind(this);
  }
  public render() {
    return (
      <Lifecycle id={this.state.id} handleInput={this.handleInput} />
    );
  }
  private handleInput(event: React.ChangeEvent<HTMLInputElement>) {
    const { value } = event.target;
    this.setState({
      id: value,
    });
  }
}

export default App;
