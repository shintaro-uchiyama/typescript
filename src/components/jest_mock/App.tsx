import React from "react";
import ExportDefault from "./export_default";
import * as ExportMulti from "./export_multi";

const App = () => {
  const stringMulti = ExportMulti.returnString;
  const functionMulti = ExportMulti.returnFunction;
  const stringDefault = ExportDefault;
  return (
    <>
      <div id="stringMulti">stringMulti: {stringMulti}</div>
      <div id="functionMulti">functionMulti: {functionMulti()}</div>
      <div id="stringDefault">stringDefault: {stringDefault}</div>
    </>
  );
};

export default App;
