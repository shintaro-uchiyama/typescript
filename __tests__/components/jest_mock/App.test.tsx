import { mount } from "enzyme";
import React from "react";
import App from "../../../src/components/jest_mock/App";
import * as ExportMulti from "../../../src/components/jest_mock/export_multi";
jest.mock("../../../src/components/jest_mock/export_default", () => {
  return "mock default string";
});

describe("App.tsx", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it("all import mock", () => {
    /** 準備 */
    const returnFunctionMock = jest.spyOn(ExportMulti, "returnFunction");
    returnFunctionMock.mockReturnValueOnce("mock multi function");
    /** 実行 */
    const wrapper = mount(<App />);
    /** 検証 */
    expect(returnFunctionMock).toBeCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
