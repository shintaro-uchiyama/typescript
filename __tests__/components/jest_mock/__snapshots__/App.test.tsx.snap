// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx all import mock 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Array [
      Object {
        "instance": <div
          id="stringMulti"
        >
          stringMulti: 
          string from multi
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "stringMulti: ",
            "string from multi",
          ],
          "id": "stringMulti",
        },
        "ref": null,
        "rendered": Array [
          "stringMulti: ",
          "string from multi",
        ],
        "type": "div",
      },
      Object {
        "instance": <div
          id="functionMulti"
        >
          functionMulti: 
          mock multi function
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "functionMulti: ",
            "mock multi function",
          ],
          "id": "functionMulti",
        },
        "ref": null,
        "rendered": Array [
          "functionMulti: ",
          "mock multi function",
        ],
        "type": "div",
      },
      Object {
        "instance": <div
          id="stringDefault"
        >
          stringDefault: 
          mock default string
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "stringDefault: ",
            "mock default string",
          ],
          "id": "stringDefault",
        },
        "ref": null,
        "rendered": Array [
          "stringDefault: ",
          "mock default string",
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            id="stringMulti"
          >
            stringMulti: 
            string from multi
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "stringMulti: ",
              "string from multi",
            ],
            "id": "stringMulti",
          },
          "ref": null,
          "rendered": Array [
            "stringMulti: ",
            "string from multi",
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            id="functionMulti"
          >
            functionMulti: 
            mock multi function
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "functionMulti: ",
              "mock multi function",
            ],
            "id": "functionMulti",
          },
          "ref": null,
          "rendered": Array [
            "functionMulti: ",
            "mock multi function",
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            id="stringDefault"
          >
            stringDefault: 
            mock default string
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "stringDefault: ",
              "mock default string",
            ],
            "id": "stringDefault",
          },
          "ref": null,
          "rendered": Array [
            "stringDefault: ",
            "mock default string",
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
